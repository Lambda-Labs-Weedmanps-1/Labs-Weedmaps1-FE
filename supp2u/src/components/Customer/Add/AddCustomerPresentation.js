import React, {useState} from "react";

export default({addCustomer, history})=>{
    let [name, setName] = useState("");

    const cancel = (e)=>{
        e.preventDefault();
        history.goBack();
    };
    const onSubmitHandler = e =>{
        e.preventDefault();
        addCustomer(name)
    };
    return (
        <form onSubmit={onSubmitHandler} className={"customer form"}>
            <input type="text" value={name} onChange={e => setName(e.target.value) } placeholder={"Name"} />
            <input type="submit" value="Create Customer"/>
            <input type="submit" value="Cancel" onClick={cancel}/>
        </form>)
}