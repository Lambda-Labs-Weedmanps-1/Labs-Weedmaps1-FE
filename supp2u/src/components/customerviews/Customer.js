import React from 'react';

const Character = ({ name, eyeColor, hairColor, gender }) => {

    return (
        <div>
            
        </div>
    );
};

export default Character;