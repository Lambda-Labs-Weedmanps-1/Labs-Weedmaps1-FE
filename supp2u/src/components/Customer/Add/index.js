import React from "react";
import {addCustomer} from "../../../actions/customerActions";
import { connect } from "react-redux";
import AddCustomerPresentation from "./AddCustomerPresentation";
import {toast} from "react-toastify";

export default connect(null, {addCustomer})(({addCustomer, history})=>{

    const addCustomerHandler = ()=>{
        let user_id=1; //TODO get user_id from localStorage
        addCustomer(user_id)
            .then((res) => {
                if(res.review){
                    toast.success("Welcome to yelp Clone");
                    history.goBack();
                }else{
                    toast.error("Oh no! something went wrong, Please try again", res.response);
                }
            });
    };
    return <AddCustomerPresentation addCustomer={addCustomerHandler} history={history}/>
})