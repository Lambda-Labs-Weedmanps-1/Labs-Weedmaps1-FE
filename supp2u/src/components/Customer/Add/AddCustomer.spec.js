import React from 'react';
import { mount, configure, shallow, enzyme } from 'enzyme';
import { MemoryRouter } from 'react-router';
import Adapter from 'enzyme-adapter-react-16';

import Customer from './AddCustomerPresentation'
import AddCustomerPresentation from './AddCustomerPresentation';

configure({ adapter: new Adapter() });

describe('AddCustomer Renders', () => {
    // If Map component render is ever non functional, this should fail
    it('should render correctly when component called', () => {
      const Customer = shallow(<AddCustomerPresentation />);

      expect(Customer).toMatchSnapshot();
    });
});