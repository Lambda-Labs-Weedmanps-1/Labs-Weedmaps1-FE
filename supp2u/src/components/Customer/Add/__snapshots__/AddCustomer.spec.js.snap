// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCustomer Renders should render correctly when component called 1`] = `
<form
  className="customer form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    type="submit"
    value="Create Customer"
  />
  <input
    onClick={[Function]}
    type="submit"
    value="Cancel"
  />
</form>
`;

exports[`Business Creator Renders should render correctly when component called 1`] = `
<form
  className="customer form"
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    placeholder="Name"
    type="text"
    value=""
  />
  <input
    type="submit"
    value="Create Customer"
  />
  <input
    onClick={[Function]}
    type="submit"
    value="Cancel"
  />
</form>
`;
